validator/client/src/test/java/tech/pegasys/teku/validator/client/ValidatorRegistratorTest.java

```diff
@@ -151,19 +158,98 @@ void registersValidators_onBeginningOfEpoch() {
         captureValidatorApiChannelCalls(2);
 
     registrationCalls.forEach(
-        registrationCall ->
-            verifyRegistrations(
-                registrationCall,
-                List.of(
-                    validator1.getPublicKey(),
-                    validator2.getPublicKey(),
-                    validator3.getPublicKey())));
+        registrationCall -> verifyRegistrations(registrationCall, allValidatorsPublicKeys));
 
     // signer will be called in total 3 times, since from the 2nd run the signed registrations will
     // be cached
     verify(signer, times(3)).signValidatorRegistration(any(), any());
   }
 
+  @TestTemplate
+  void cleanupsCache_ifValidatorIsNoLongerActive() {
+    // GIVEN
```

Comment:
while i do understand the 'given/when/then', it does get distracting to have the comments scattered everywhere...

a couple of things, if the first bit is setup, break it out as a function to prepare the test... also make sure the test is testing 1 thing... it sounds like we're wanting the test to do a bunch of things from the 'given/when/then'

from the other comment - given/when is basically pre-test. then is the test, so basically this test may have too many test conditions.

Reply:
Thanks. I agree about the test being too verbose. Will look into ways of simplifying it and make it more readable.